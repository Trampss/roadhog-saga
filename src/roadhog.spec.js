/* eslint-env jest */
import tester from 'trampss-redux-saga-tester'

jest.mock('./tracer')
jest.mock('./checkers', () => ({
  isAction: () => true,
}))
jest.mock('./mappers', () => ({
  mapToFetch: () => () => ['http://an-url', { some: 'options' }],
}))

const tracer = require('./tracer').default
const roadhog = require('./roadhog').default

describe('roadhog', () => {
  const test = tester(roadhog('GET_TODOS'))(/* no inputs */)

  it('should return API returns (ok case)', () => {
    tracer.mockImplementation(() => () => ({ ok: true, json: () => ({ some: 'value' }) }))

    expect(
      test({
        select: [() => 'fallback'],
      }),
    ).toMatchSnapshot()
  })

  it('should return fallback (ko case)', () => {
    tracer.mockImplementation(() => () => ({ ok: false }))

    expect(
      test({
        select: [() => 'fallback'],
      }),
    ).toMatchSnapshot()
  })
})
