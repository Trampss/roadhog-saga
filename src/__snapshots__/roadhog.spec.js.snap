exports[`roadhog should return API returns (ok case) 1`] = `
Array [
  Object {
    "done": false,
    "value": Array [
      "http://an-url",
      Object {
        "some": "options",
      },
    ],
  },
  Object {
    "raw": Object {
      "done": false,
      "value": Object {
        "@@redux-saga/IO": true,
        "SELECT": Object {
          "args": Array [],
          "selector": [Function],
        },
      },
    },
    "value": "fallback",
  },
  Object {
    "done": false,
    "value": Object {
      "json": [Function],
      "ok": true,
    },
  },
  Object {
    "done": false,
    "value": Object {
      "some": "value",
    },
  },
  Object {
    "done": true,
    "value": Object {
      "some": "value",
    },
  },
]
`;

exports[`roadhog should return fallback (ko case) 1`] = `
Array [
  Object {
    "done": false,
    "value": Array [
      "http://an-url",
      Object {
        "some": "options",
      },
    ],
  },
  Object {
    "raw": Object {
      "done": false,
      "value": Object {
        "@@redux-saga/IO": true,
        "SELECT": Object {
          "args": Array [],
          "selector": [Function],
        },
      },
    },
    "value": "fallback",
  },
  Object {
    "done": false,
    "value": Object {
      "ok": false,
    },
  },
  Object {
    "done": true,
    "value": "fallback",
  },
]
`;
